llvm/lib/MC/MCAssembler.cpp

```diff
@@ -777,6 +777,14 @@ bool MCAssembler::relaxInstruction(MCFragment &F) {
   getEmitter().encodeInstruction(Relaxed, Data, Fixups, *F.getSubtargetInfo());
   F.setVarContents(Data);
   F.setVarFixups(Fixups);
+
+  for (const auto &Fixup : Fixups) {
+    if (!Fixup.isLinkerRelaxable())
+      continue;
+    F.setLinkerRelaxable();
+    F.getParent()->setLinkerRelaxable();
+  }
```

Comment:
This does not cope with the other direction correctly, but I'm not sure we can.

The other direction is when a linker relaxable fixup is replaced by a non-linker-relaxable fixup in relaxation, which might need the fragment to be marked as not linker relaxable. I'm not sure this will ever happen though.